hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -0,0 +1,251 @@
+import com.hedera.node.app.service.mono.state.migration.HederaAccount;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.Key;
+import com.hederahashgraph.api.proto.java.KeyList;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bouncycastle.asn1.sec.SECNamedCurves;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.crypto.params.ECPublicKeyParameters;
+import org.bouncycastle.util.encoders.DecoderException;
+import org.bouncycastle.util.encoders.Hex;
+
+/**
+ * Encapsulates the logic for reading blocked accounts from file and creating them.
+ */
+@Singleton
+public class BlocklistAccountCreator {
+    private static final int ZERO_BALANCE = 0;
+    private static final Logger log = LogManager.getLogger(BlocklistAccountCreator.class);
+    private final String blocklistFileName;
+    private final Supplier<HederaAccount> accountSupplier;
+    private final EntityIdSource ids;
+    private final BackingStore<AccountID, HederaAccount> accounts;
+    private final Supplier<JEd25519Key> genesisKeySource;
+    private final PropertySource properties;
+    private final AliasManager aliasManager;
+    private JKey genesisKey;
+    private final List<HederaAccount> blockedAccountsCreated = new ArrayList<>();
+    private AccountNumbers accountNumbers;
+
+    /**
+     * @param evmAddress the EVM address of the blocked account
+     * @param memo      the memo of the blocked account
+     */
+    record BlockedInfo(ByteString evmAddress, String memo) {}
+
+    @Inject
+    public BlocklistAccountCreator(
+            final Supplier<HederaAccount> accountSupplier,
+            final EntityIdSource ids,
+            final BackingStore<AccountID, HederaAccount> accounts,
+            final Supplier<JEd25519Key> genesisKeySource,
+            final @CompositeProps PropertySource properties,
+            final AliasManager aliasManager,
+            AccountNumbers accountNumbers) {
+        this.blocklistFileName = properties.getStringProperty(BLOCKLIST_FILE);
+        this.accountSupplier = accountSupplier;
+        this.ids = ids;
+        this.accounts = accounts;
+        this.genesisKeySource = genesisKeySource;
+        this.properties = properties;
+        this.aliasManager = aliasManager;
+        this.accountNumbers = accountNumbers;
+    }
+
+    /**
+     * Makes sure that all blocked accounts contained in the blocklist file are present in state, and creates them if necessary.
+     */
+    public void ensureBlockedAccounts() {
```

Comment:
nit: I think it's more true to say this method creation accounts that don't already exist in state.
```suggestion
    public void createMissingAccounts() {
```

Reply:
Updated.